dropwizard-testing/src/main/java/io/dropwizard/testing/DropwizardTestSupport.java

```diff
@@ -150,7 +185,15 @@ public void serverStarted(Server server) {
                     }
                 }
             };
-            if (customPropertyPrefix.isPresent()) {
+            if (explicitConfig) {
+                bootstrap.setConfigurationFactoryFactory(new ConfigurationFactoryFactory<C>() {
+                    @Override
+                    public ConfigurationFactory<C> create(Class<C> klass, Validator validator,
+                                                          ObjectMapper objectMapper, String propertyPrefix) {
+                        return new POJOConfigurationFactory<C>(configuration);
```

Comment:
We compile against Java 7, so we could use just `new POJOConfigurationFactory<>(configuration)`